### DISCUSSION TITLE
How to fill specific channel with either black or white ?

### DISCUSSION THREAD

**Discussion by TheSpartaPT** [ImageMagick/ImageMagick#4334] [Help] [answered]
I've tried both:

magick input.png -channel R -fill '#FFFFFF' output.png
magick input.png -channel R -fill white output.png

they don't fill the red channel white, i'm probably doing something wrong or using the wrong cli options but i can't find any way to do this

**Comment by fmw42** [comment, marked as answer]
-fill is a setting, not an operator. It needs something to process. So instead, just set the red channel to full gray level and then turn back on the other channels.

Try

`magick input.png -channel r -evaluate set 100% +channel output.png`

**Reply by TheSpartaPT** [reply]
yup that did it, thanks 😄
